### PR TITLE
Authentication with Devise

### PR DESCRIPTION
**Pull request by tobuya**
# Using Devise for user authentication

The following changes are made in this branch:
- Generated `views` and `model` with `devise`
- Created registration parameters in `ApplicationController`
- Implemented authentication features
- Modified views and layouts

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
Comment (addod19) [inline app/models/user.rb]: - [ ] It would be better if we enable `confirmable` so that users will have to confirm when registering to the application

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @tobuya and @IndieCoderMM, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [ ] Correct Git-flow followed :+1:
- [ ] No linter errors recorded :100: 
- [ ] Devise used and installed :rocket: 
- [ ] Awesome collaboration, the best I have seen so far :man_technologist: 

## Required Changes ♻️
_Check the comments under the review._
- [ ] It would be better if we fix our MIT LICENSE in our README file, also if we have not deployed the app, let us kindly remove the misleading link

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
- [ ] Going forward, let us capitalized all our commit messages :+1: 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (tobuya) [reply app/models/user.rb]: @addod19 Thank you for the suggestions, I'm working on them

Comment (rloterh) [reply app/models/user.rb]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Muller,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 👍
- Great job implementing recommended changes!

Cheers..Happy coding! 💻  🍷  🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**